nav2_costmap_2d/plugins/costmap_filters/binary_filter.cpp

```diff
@@ -84,6 +107,19 @@ void BinaryFilter::initializeFilter(
     filter_info_topic_, rclcpp::QoS(rclcpp::KeepLast(1)).transient_local().reliable(),
     std::bind(&BinaryFilter::filterInfoCallback, this, std::placeholders::_1));
 
+  // Create clients for changing parameters
+
+  for (auto param : binary_parameters_info_) {
+    RCLCPP_DEBUG(
+      logger_,
+      "BinaryFilter: Creating client for changing parameter \"%s\" of node \"%s\"...",
+      param.param_name.c_str(), param.node_name.c_str());
+
+    auto change_parameters_client_ = node->create_client<rcl_interfaces::srv::SetParameters>(
+      "/" + param.node_name + "/set_parameters");
```

Comment:
I can removed the absolute slash (`"/" +`) to make it automatically handle namespaces, but I guess then there would be an issue with `/local_costmap`/`/global_costmap` sub-namespace of the costmap, which - I think - is the same reason why you use `/scan` (instead of scan) in the obstacle layers (using a relative source, e.g. `scan` creates a subscription to `/global_costmap/scan`). 

I mean, I kept it absolute to have more control. This way a user can add the namespace in the params file, while if the namespace would be handled automatically a user would need to do a remap.
What do you suggest?
I think something like this (from[ costmap_2d_ros](https://github.com/ros-planning/navigation2/blob/d9bea52838d67b7ecfefb60788a3a4ff61f587ca/nav2_costmap_2d/src/costmap_2d_ros.cpp#L103) would solve it:
```
     "map_topic", rclcpp::ParameterValue(
      (parent_namespace_ == "/" ? "/" : parent_namespace_ + "/") + std::string("map")));

```

But I don't have such information (`parent_namespace_`) at costmap filter level

Reply:
To be honest I'm not sure about how I could solve this... As I said, the filter has not the access level to distinguish between the two namespaces, so only workarounds come to my mind (e.g. the only two sub_namespace I can image for these filter are `global_costmap` and `local_costmap`, so I could simply cut them from the service name), but I don't think that  having earthbound reasoning is acceptable

Reply:
What will be happened, if you will just remain the
```
 auto change_parameters_client_ = node->create_client<rcl_interfaces::srv::SetParameters>(
      param.node_name + "/set_parameters");
```
without forwarding "/"?

- If `param.node_name` is specified in absolute path (e.g. `/some_node`) will be will applied as `/some_node/set_parameters` service.
- If `param.node_name` is specified in relative path (e.g. `some_node`), won't service will be created relative to current BinaryFilter namespace: `some_node/set_parameters` which will be treated as `/my_namespace/some_node/set_parameters` when operating with it in run-time?